lib/storage/data/LocationConstraintParser.js

```diff
@@ -25,21 +26,26 @@ function parseLC(config, vault) {
         if (locationObj.type === 'file') {
             clients[location] = new DataFileBackend(config);
         }
-        if (locationObj.type === 'scality'
-        && locationObj.details.connector.sproxyd) {
-            clients[location] = new Sproxy({
-                bootstrap: locationObj.details.connector
+        if (locationObj.type === 'scality') {
+            if (locationObj.details.connector.sproxyd) {
+                clients[location] = new Sproxy({
+                    bootstrap: locationObj.details.connector
                     .sproxyd.bootstrap,
-                // Might be undefined which is ok since there is a default
-                // set in sproxydclient if chordCos is undefined
-                chordCos: locationObj.details.connector.sproxyd.chordCos,
-                // Might also be undefined, but there is a default path set
-                // in sproxydclient as well
-                path: locationObj.details.connector.sproxyd.path,
-                // enable immutable optim for all objects
-                immutable: true,
-            });
-            clients[location].clientType = 'scality';
+                    // Might be undefined which is ok since there is a default
+                    // set in sproxydclient if chordCos is undefined
+                    chordCos: locationObj.details.connector.sproxyd.chordCos,
+                    // Might also be undefined, but there is a default path set
+                    // in sproxydclient as well
+                    path: locationObj.details.connector.sproxyd.path,
+                    // enable immutable optim for all objects
+                    immutable: true,
+                });
+                clients[location].clientType = 'scality';
```

Comment:
Minor, but this line could be pulled out since it's included in both Sproxyd and Hyperdrive conditionals

Reply:
May be we should introduce `hyperdrive` as a type. I think back in the day we used `scality` as we didn't envision using any other connector in Scality.